dashboard/lib/devise/models/custom_lockable.rb

```diff
@@ -32,46 +32,54 @@ def increment_failed_attempts
         reload
       end
 
-      # Record lock and unlock events.
+      # Record lock and unlock events. Make sure to only log method invocations
+      # that will actually result in a state change, to avoid logging no-op
+      # invocations such as from the password reset workflow.
       #
       # As a core system event related to the user model, we want to log these
-      # in CloudWatch in order to preverse metrics for the long term. And as a
+      # in CloudWatch in order to preserve metrics for the long term. And as a
       # newly-enabled feature, we want to also log them in Statsig to make the
       # metric available on the product team's dashboards in the short term.
 
       # @override https://github.com/heartcombo/devise/blob/v4.9.3/lib/devise/models/lockable.rb#L42-L50
       def lock_access!
-        # CloudWatch
-        Cdo::Metrics.put(
-          'User', 'DeviseLockableAccessLocked', 1, {
-            Environment: CDO.rack_env,
-            UserType: user_type
-          }
-        )
+        unless access_locked?
```

Comment:
We aren't currently logging any excessive lock events, only excessive unlock events, so there's some extent to which it's unnecessary to update this method. But I included it for both parallelism with `unlock_access!` and resiliency against future changes.